[Merged] 1696 debug report

yuriikarpenkoo opened the pull request.

sschiessl-bcp commented on app/AppInit.jsx:
```diff
@@ -54,7 +55,34 @@ class AppInit extends React.Component {
+        const saveLog = (type, log) =>
+            LogsActions.setLog({type, log: Array.from(log)});
+
+        console.log = function() {
```
The overwritten console function should still print to the console log while also filling up the new log storage

sschiessl-bcp commented on app/actions/AccountActions.js:
```diff
@@ -25,12 +25,18 @@ class AccountActions {
+                            "We have error with accountSearch mathod: ",
+                            error
+                        );
+                        reject(error);
```
There is no reject method in this scope.

This is an indirect search. Consider for example AccountNameInput class
https://github.com/blockchainprojects/bitshares-ui/blob/c5016f8a0b15d5b0dedc51b1d69ad4a07f1a8018/app/components/Forms/AccountNameInput.jsx#L11

It triggers the account search via the action
https://github.com/blockchainprojects/bitshares-ui/blob/c5016f8a0b15d5b0dedc51b1d69ad4a07f1a8018/app/components/Forms/AccountNameInput.jsx#L124
This return value of this action is not caught, but dispatched back into the component via AccountStore (this is the alt store mechnism) once the AccountActions.accountSearch Promise has resolved
https://github.com/blockchainprojects/bitshares-ui/blob/c5016f8a0b15d5b0dedc51b1d69ad4a07f1a8018/app/components/Forms/AccountNameInput.jsx#L187-L191

For the example above, please change it that there is a console error call with a precise message, e.g. in the style of
`console.error("Exception in AccountActions.accountSearch(", start_symbol, ", ", limit, " occured: ", err)`

Please check all similar occurrences in introduced ".catch" methods for Promises throughout this PR

sschiessl-bcp commented on app/assets/locales/locale-fr.json:
```diff
@@ -1177,6 +1177,9 @@
             "submit_beta": "Withdraw (beta)",
             "withdraw": "Withdraw"
+        },
+        "report": {
```
Forgot the other keys

sschiessl-bcp commented on app/components/Blockchain/MemoText.jsx:
```diff
@@ -85,13 +85,16 @@ class MemoTextStoreWrapper extends React.Component {
-        return {
-            wallet_locked: WalletUnlockStore.getState().locked
-        };
+export default connect(
```
Please remove all changes that only update formatting from this PR

yuriikarpenkoo committed:
1696 Removed impacted style

yuriikarpenkoo committed:
reject fix

sschiessl-bcp commented:
Open e.g. market/USD_BTS and observe the log filling, but debug report does not contain those messages

sschiessl-bcp commented:
The Debug Report modal says "Memo / Message". Rather have "Last 20 log entries" or similar

VerevkinAlexander committed:
Fixed issue with incorrect logs form title and missing logs

VerevkinAlexander committed:
Refactored code , changed buttons logic and position

sschiessl-bcp commented:
Bug in log state on startup
![image](https://user-images.githubusercontent.com/33128181/46455056-8df76000-c7a9-11e8-9ca9-e8ab1a47ea43.png)

Can you please elaborate the intent of `extendeLogText` as a state variable? Since it is persisted in localStorage.

VerevkinAlexander commented:
> Can you please elaborate the intent of `extendeLogText` as a state variable? Since it is persisted in localStorage.

It is work as a buffer. If an error occurs in the node-module local storage stops writing log on some time and node-module error and another one, after that they are missing in debug log.

VerevkinAlexander committed:
Fixed Bug in log state on startup

sschiessl-bcp commented on app/actions/AccountActions.js:
```diff
@@ -87,11 +87,18 @@ class AccountActions {
+
+                        dispatch(result);
+                    })
+                    .catch(error => {
```
WIth the global componentDidCatch, are individual catches necessary? The unresolved Promise exception will be caught, right?

I understand it's best practice to do so, but I am afraid we might accidentally change logic somewhere.

VerevkinAlexander committed:
Deleted new .catch()

Commit:
minor styling

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

sschiessl-bcp commented:
It seems that your internal design creates recursion loops. Please place a 
```
throw "Some error"
```
in the render of any component, e.g. the Footer.jsx.

I then see this error 
![image](https://user-images.githubusercontent.com/33128181/46594848-ba76e900-cad5-11e8-8214-000e346c0544.png)

It multiplies until stack is full und ultimately the UI crashes completely.

VerevkinAlexander committed:
Deleted reportActions and Reportstore

VerevkinAlexander committed:
Merge branch '1696_DebugReport' of https://github.com/blockchainprojects/bitshares-ui into 1696_DebugReport

VerevkinAlexander committed:
Fixed issue with loop

VerevkinAlexander committed:
refactored code

sschiessl-bcp commented on app/assets/locales/locale-ja.json:
```diff
@@ -1641,7 +1650,7 @@
         "header": "転送の詳細設定",
         "header_subheader": "転送機能は他のBitSharesアカウントに資金を送金するために使用されます。",
-        "memo": "メモ",
+        "memo": "最後の20のログエントリ",
```
I did not see that before. Please refactor that you do not use news.memo (since this is used elsewhere as well).

Please create a proper entry within your new report section (for all languages)

sschiessl-bcp commented:
Code looks good now. Did you incorporate to copy the Screenshot into Clipboard?

VerevkinAlexander commented:
> Code looks good now. Did you incorporate to copy the Screenshot into Clipboard?

Not yet. Now, I am trying to implement it. I found that html2canvas come out from the user click process and "Copy" command cannot be done

VerevkinAlexander committed:
Added screen copy

VerevkinAlexander commented:
I found that there is no acceptable way to copy the image to the clipboard because browsers prevent any process of copying files that was not provided via user. I added preview that can be copyed via user by right click.

sschiessl-bcp commented:
How did you implemenent the copy?

I thougt it would be user initiliazed. He clicks "Copy screenshot" or similar, and gets it into clipboard

VerevkinAlexander commented:
- I tried to create a new image and append it to the body, then selected it and call execCommand("Copy") this command response is successful but clipboard is empty( if I used text instead of the image its copied successfully).
- I  tried to intercept user copy action and redefined it. In the redefined copy action, I called clipboardData.setData and throw image to it.
- I  tried to use webApi but also no luck 
In all this cases copying status was successful but clipboard was empty

VerevkinAlexander committed:
Refactored

Commit:
- do not show html2canvas logs
- add description text
- ReportModal Rrefactoring
  - memo -> logEntries
  - show preview of screenshot
  - restyle the component
  - remove code that is copy-paste leftover ({/*  M E M O  */} etc.)

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

VerevkinAlexander committed:
refactored news.memo

VerevkinAlexander committed:
Merge branch '1696_DebugReport' of https://github.com/blockchainprojects/bitshares-ui into 1696_DebugReport

VerevkinAlexander committed:
refactored report.modal.title

sschiessl-bcp commented on app/actions/LogsActions.js:
```diff
@@ -0,0 +1,14 @@
+
+class LogsActions {
+    async setLog(log) {
+        return await localStorage.setItem("logs", JSON.stringify(log));
```
Don't use localStorage directly, as it might not exist (see localStorageImpl). Use app/lib/common/localStorage.

VerevkinAlexander replied:
fixed

sschiessl-bcp commented on app/lib/common/Screenshot.js:
```diff
@@ -0,0 +1,26 @@
+    const width = window.innerWidth / typeSize;
+    const height = window.innerHeight / typeSize - footerHeight;
+
+    html2canvas(content).then(canvas => {
```
Is this still used with the screenshot being directly shown in the modal? If not please remove, I assume we also don't need the pdf creation dependencies anymore then.

I will have an issue for PDF creation soon for you, but not within this one.

VerevkinAlexander replied:
fixed

VerevkinAlexander committed:
Deleted Screenshot and changed direct LocalStorage to common LocalStorag

Commit:
Merge branch 'develop' into 1696_DebugReport

Commit:
cleanup unused, and format
cleanup package

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

Commit:
revert catch, not in scope
only call _enablePersistingLog once

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

Commit:
revert package lock

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

Commit:
Merge branch 'develop' into 1696_DebugReport

# Conflicts:
#	app/App.jsx
#	app/assets/stylesheets/themes/_theme-template.scss

Commit:
remove locales besides engflish

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

sschiessl-bcp commented:
Accepted, will forward
